### PR TITLE
Use datasource cloudant

### PR DESCRIPTION
**Pull request by jannyHou**
#### Related issues
Connect to strongloop-internal/scrum-apex#155

I create a new branch `use-datasource-cloudant` to demo access control(Role and User) with relevant models attached to cloudant datasource.

There are some syntax changes in this PR because master is still using an old version of loopback. The three main changes are:

- server/datasources.json
    - Add cloudant datasource
- server/model-config.json
    - Attach relevant models to cloudant.
- server/boot/role-resolver.js
    - automigrate models attached to cloudant datasource    
    - Switch to async boot script, since other scripts need to wait until automigrating models to cloudant datasource done.

And udpated the doc as well.

### CODE REVIEW

File: common/models/user.json [hunk 1]
```diff
@@ -2,10 +2,13 @@
...
   "base": "User",
   "idInjection": true,
+  "options": {
+    "validateUpsert": true
+  },
   "properties": {},
   "validations": [],
   "relations": {
-    "projects": {
+     "projects": {
```
Comment (0candy) [inline common/models/user.json]: Extra space not needed here. Please revert.

File: README.md [hunk 2]
```diff
@@ -48,19 +49,31 @@ $ lb app loopback-example-access-control
 ... # follow the prompts
 $ cd loopback-example-access-control
 ```
+### Add the datasource
+- Name: `mycloudant`
+  - datasource: `cloudant`
+  - url: `your_cloudant_url`
+  - database: `your_cloudant_database`
+  - username: `your_cloudant_username`
```
Comment (dhmlau) [inline README.md]: do we need username and password?  i thought it's already sufficient if we have url.

File: README.md [hunk 3]
```diff
@@ -1,9 +1,10 @@
-# loopback-example-access-control
+# loopback-example-access-control(use datasource cloudant)
 
 ```
 $ git clone https://github.com/strongloop/loopback-example-access-control
 $ cd loopback-example-access-control
 $ npm install
+Manually change credential of `mycloudant` in server/datasources.json to your cloudant credential
```
Comment (dhmlau) [inline README.md]: Suggested change (i don't think we need the word "manually" :) ): 
Go to server/datasources.json and change the credential of `mycloudant` to your cloudant credential.

File: README.md (hunk 2 above)
Comment (jannyHou) [inline README.md]: True, while some user doesn't have url so they provide usr&pwd, I will link to cloudant#config here so they won't get confused: https://github.com/strongloop/loopback-connector-cloudant#configuration

**Comment by jannyHou** [comment]
@0candy @dhmlau thanks, feedback applied.

**Comment by ssh24** [comment]
@jannyHou Are the CI failures well taken note of for why is it failing?

**Comment by jannyHou** [comment]
@ssh24 sorry I missed your question:
CI fails because I use a fake url in example and loopback-connector-cloudant@2.x is not released.

**Comment by jannyHou** [comment]
Cloudant@2.x is released but test fails due to missing a valid database to run tests....I need to setup a docker container

**Comment by dhmlau** [comment]
@jannyHou , what's the outstanding items to get this landed? thanks.

**Pull request merged**